src/improvements/template/boilerplate/OPCMTaskBase.template.sol

```diff
@@ -81,8 +81,11 @@ contract OPCMTaskBaseTemplate is OPCMTaskBase {
     /// - `Multicall3Delegatecall`:
     ///   If the template inherits from `OPCMTaskBase`, it uses the `Multicall3Delegatecall` contract.
     ///   In this case, calls to the target **must** use `delegatecall`, e.g.:
-    ///   `(bool success,) = OPCM.delegatecall(abi.encodeWithSelector(IOPContractsManager.upgrade, opChainConfigs));`
+    ///   `(bool success,) = OPCM.delegatecall(abi.encodeWithSelector(IOPContractsManager.upgrade, opChainConfigs));
+    /// WARNING: Any state written to in this function will be reverted after the build function has been run.
+    /// Do not rely on setting global variables in this function.
```

Comment:
good safety feature to track is to assert this. Implementation could be something like:
- `vm.dumpState()` -> filter for storage in `address(this)`
- do the build method
- dump state again after build and `keccak256(initStateOfThis)) == keccak256(endStateOfThis)`